compiler/rustc_mir_transform/src/coverage/spans.rs

```diff
@@ -9,12 +12,18 @@ use crate::coverage::ExtractedHirInfo;
 
 mod from_mir;
 
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Debug)]
 pub(super) enum BcbMappingKind {
     /// Associates an ordinary executable code span with its corresponding BCB.
     Code(BasicCoverageBlock),
     /// Associates a branch span with BCBs for its true and false arms.
-    Branch { true_bcb: BasicCoverageBlock, false_bcb: BasicCoverageBlock },
+    Branch {
+        true_bcb: BasicCoverageBlock,
+        false_bcb: BasicCoverageBlock,
+        condition_info: ConditionInfo,
+    },
```

Comment:
Similarly, I don't think we should add MC/DC information to the existing `Branch` variant. Instead, we should have separate variants for plain branches and MC/DC branches.

Reply:
Done, but the code at generating BcbMappingKinds could be improved? Currently looks a bit ugly.